File: examples/dart/spin_bot.dart
import 'package:codetanks/codetanks.dart';

class MyTank extends BaseTank {
  int i = 0;

  MyTank() {
    print("Created my spin tank!");
  }

  @override
  void run() {
    if (i % 2 == 0) {
      commands.add(Command.MOVE_FORWARD | Command.ROTATE_TANK_CLOCKWISE | Command.FIRE);
    } else {
      commands.add(Command.MOVE_BACKWARD | Command.ROTATE_TANK_COUNTER_CLOCKWISE | Command.FIRE);
    }    
    print(commands);
  }
  
  @override
  void onEvent(Map event) {
    print(event);
    i = i + 1;
    print(i);
  }
}

BaseTank createTank() => MyTank();

File: examples/python/spin_bot.py
from codetanks import BaseTank, commands

class MyTank(BaseTank):
    def __init__(self):
        super().__init__()

        self.direction = 0

        print('Running my spinning tank!')

    def run(self):
        if self.direction % 2 == 0:
            self.commands.append(commands.MOVE_BACKWARD | commands.ROTATE_TANK_COUNTER_CLOCKWISE | commands.FIRE)
        else:
            self.commands.append(commands.MOVE_FORWARD | commands.ROTATE_TANK_CLOCKWISE | commands.FIRE)

    def on_event(self, event):
        self.direction = self.direction + 1

def create_tank():
    return MyTank()